presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoConfigDeployer.java

```diff
@@ -49,54 +48,78 @@
         this.executor = requireNonNull(executor);
     }
 
-    public File storeConfigFiles(Path configDir)
+    /**
+     * Backup the given directory in a temporary file.
+     * <p>
+     * The only guarantee about the returned {@link Path} is that it may be
+     * restored with {@link #restoreDirectory(Path, Path)}.
+     * <p>
+     * The backup file may be created in the system's temporary file directory.
```

Comment:
Using "may" allows us to change that in the future if we want. This way, nobody programming around this should make any assumptions about the backup's location.